common/models/transactions.py

```diff
@@ -41,6 +48,35 @@ def ordered_tracked_models(self):
         )  # order_by record_code, subrecord_code already happened in get_queryset
         return tracked_models
 
+    def last_finalised_transaction(self):
+        return self.filter(status=TransactionStatus.FINALISED).order_by("order").last()
+
+    def finalise_transactions(self):
+        """Set transaction status to FINALISED and ordering to global."""
+        statuses = self.distinct("status").values_list("status", flat=True)
+
+        if statuses == [TransactionStatus.FINALISED.value]:
+            logger.info("Transactions are already finalised.")
+            return
+
+        if statuses != [TransactionStatus.DRAFT.value]:
+            logger.warning("Transactions have non-DRAFT statuses: %s", statuses)
+            return
+
+        with atomic():
+            workbaskets = self.values("workbasket").distinct()
+            last_tx = self.model.exclude(
+                workbasket__in=workbaskets,
+            ).last_finalised_transaction()
+            last_order = last_tx.order if last_tx is not None else 0
+            for global_id, txn in enumerate(
+                self.order_by("order"),
+                last_order + 1,
+            ):
+                txn.status = TransactionStatus.FINALISED
+                txn.order = global_id
+                txn.save()
```

Comment:
Would it be better to use [`QuerySet.update`](https://docs.djangoproject.com/en/3.2/ref/models/querysets/#django.db.models.query.QuerySet.update)?

```suggestion
            self.update(
                order=F("order") + Value(last_order),
                status=TransactionStatus.FINALISED,
            )
```

Reply:
Thanks, I was blind to how this would work with an update :)

Reply:
OK, sorted.